Extension/src/extensionActivationInformation.ts

```diff
@@ -0,0 +1,44 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All Rights Reserved.
+ * See 'LICENSE' in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+
+export enum InstallBlobStage {
+    getPlatformInfo,
+    downloadPackages,
+    installPackages,
+    makeBinariesExecutable,
+    makeOfflineBinariesExecutable,
+    removeUnnecessaryFile,
+    touchInstallLockFile,
+    rewriteManifest,
+    postInstall
+}
+
+export class InstallBlob {
+    stage: InstallBlobStage;
+    hasError: boolean;
+    telemetryProperties: { [key: string]: string };
+
+    constructor() {
+        this.hasError = false;
+        this.telemetryProperties = {};
+    }
+}
+
+let installBlob: InstallBlob;
```

Comment:
I am of the opinion that the installation state should be managed within a single module, not decentralized across multiple modules.  Can you please keep the object inside main.ts and remove these exports?  Or was there something that made it hard to call setInstallBlobStage from within the flow of events in main.ts?

Reply:
I personally felt that it made sense that the function that was being called would set the InstallBlobStage rather than the caller. E.g.

```
installBlob.stage = "downloadPackages"; 
packageManager.DownloadPackages(); 
...
installBlob.stage = "installPackages"; 
packageManager.InstallPackages(); 
```

Reply:
The caller sets the stage in every other instance.  I think that any possible confusion is caused by the fact that two "stages" occur in `downloadAndInstallPackages` instead of splitting this function into two (e.g. `downloadPackages` and `installPackages`) and chaining them.  If you did that then the pattern would look consistent across the entire file.